### PR TITLE
Add events to calendar (fitness classes, academic & career events)

### PR DESCRIPTION
**Pull request by huangshawn1**
Allows users to tap fitness class cells, academic cells, or buttons in career event details to add an event to their Apple calendar.

- EventManager: class to manage all interaction with user's calendar, set parameters for added event
- CalendarEvent
  - additionalDescription: allows different classes to include different elements in the notes/description section of the event. this appears in addition to the "description" string. (e.g. trainer, register link)
  - addToDeviceCalendar: prompts the user with an alert if they want to add the event to their calendar

Also removed fitness class upcoming card and changed the "Today" card to show all future classes.
Added message to display when requesting calendar permission, modified location access message to better reflect what location is used for.

Screenshots:
IMG_1836
IMG_1838
IMG_1839
IMG_1840

### CODE REVIEW

**Comment by Stereopoly** [comment]
Lgtm! I think it's slightly on the fast side though, maybe increase duration slightly

**Review by KevWho — CHANGES_REQUESTED** [review]
Works well, changes look good! Talked with Sud briefly about getting a "full-day" field for the events that last all day, so hopefully those events will be better added to calendars in the future.

File: berkeley-mobile/Fitness/FitnessViewController.swift [hunk 1]
```diff
@@ -301,6 +242,13 @@ extension FitnessViewController {
     
 }
 
+extension FitnessViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        futureClasses[indexPath.row].addToDeviceCalendar(vc: self)
+        tableView.deselectRow(at: indexPath, animated: false)
```
[the remark this answers is not in the document]
Comment (huangshawn1) [reply berkeley-mobile/Fitness/FitnessViewController.swift:248]: set to true because it's true everywhere else in the app

**Review by KevWho — APPROVED** [review]
Looks good to me! @Stereopoly looks like the duration for the success alert was increased to 0.7, does this address your comment?

**Pull request merged**